File: frontend/src/models/alert.ts
export type AlertLevel = "bloquante" | "analyser" | "informative"
export type AlertStatus = "nouvelle" | "en_cours" | "cloturee" | "classee"

export interface Alert {
  id: string
  ref: string
  clientId: string
  client: string
  score: number
  type: string
  level: AlertLevel
  module: string
  facteurs: string[]
  status: AlertStatus
  analyste: string
  createdAt?: string
}

export interface AlertFilter {
  status?: string
  level?: string
  module?: string
  analyste?: string
}

File: backend/app/schemas/alert.py
from typing import Optional, List, Any
from datetime import datetime
from pydantic import BaseModel, ConfigDict


class AlertBase(BaseModel):
    reference: str
    client_id: str
    type_alerte: str
    niveau: str = "analyser"  # bloquante, analyser, informative
    score: int = 0
    module: str = "Risk Score"
    statut: str = "nouvelle"
    analyste: Optional[str] = None
    facteurs: List[Any] = []


class AlertCreate(AlertBase):
    pass


class AlertUpdate(BaseModel):
    statut: Optional[str] = None
    analyste: Optional[str] = None
    niveau: Optional[str] = None


class AlertOut(AlertBase):
    id: str
    created_at: datetime
    client_nom: Optional[str] = None
    client_code: Optional[str] = None

    model_config = ConfigDict(from_attributes=True)
